src/OpenTelemetry/Trace/Export/SpanData.cs

```diff
@@ -66,19 +58,18 @@ public struct SpanData
             Status status,
             DateTimeOffset endTimestamp)
         {
-            this.span = null;
-
-            this.name = name;
-            this.context = context;
-            this.parentSpanId = parentSpanId;
-            this.kind = kind;
-            this.startTimestamp = startTimestamp;
-            this.attributes = attributes;
-            this.links = links;
-            this.events = events;
-            this.resource = resource;
-            this.status = status;
-            this.endTimestamp = endTimestamp;
+            this.span = new SpanSdk(
+                name,
+                context,
+                parentSpanId,
+                kind,
+                startTimestamp,
+                attributes,
+                events, links,
+                resource,
+                status,
+                endTimestamp,
+                DefaultConfiguration);
```

Comment:
Should we be carful about using a default here?

Reply:
yeah, good point. I was weighting different options.
For any legit use of OTel, default configuration should be reasonable and should not be an issue (it just limits number of attributes to 32, events to 128, links to 32). 

Configuration provides safe limits on the links/attributes/events count and the requirement to have them is to prevent  unlimited consumption of memory by OTel. [Spec does not yet](https://github.com/open-telemetry/opentelemetry-specification/issues/182) provide expected behavior regarding these limits. Current implementation is inherited from OpenCensus/OTel Java.

One concern I had is that if someone called SpanData with enormous list of events or attributes, OTel limitations would not really help the app, but at least OTel would not be the one responsible for the memory consumption :)

I guess what we can do is expose static TracerConfiguration propoerty with public setter on SpanData with default value. 

Alternative is to wait for the spec to formalize requirements what behavior is expected and which limits should be used and whether they have to be configurable at all.

What do you think?

Reply:
I think we can leave it as is with the `DefaultConfiguration` and wait for spec to finalise. We can always go back to add static setter if it becomes an issue.